src/cobra/io/mat.py

```diff
@@ -179,6 +353,214 @@ def create_mat_metabolite_id(model: Model) -> str:
+                if x and len(_pumbed_re.sub("", x).strip())
+                else None
+                for x in _notes
+            ]
+        elif mat_key == "rxnConfidenceScores":
+            # If it something else, which may have commas, turn it into a list
+            notes[name] = [
+                str(confidence)
+                for confidence in _cell_to_float_list(mat_struct[mat_key][0, 0])
+            ]
+        else:
+            # If it something else, which may have commas, turn it into a list
+            notes[name] = _cell_to_str_list(mat_struct[mat_key][0, 0])
+    for i, obj in enumerate(target_list):
+        obj.notes = {prov: notes[prov][i] for prov in note_providers if notes[prov][i]}
+
+
+def annotations_to_mat(
+    mat_dict: OrderedDict, annotation_list: List[Dict], _d_replace: str = D_MET_REV
+) -> None:
+    """Process mat structure annotations in place.
+
+    Will process mat structured annotations and add them to a list of new entities
+    (metabolites, reactions, genes) in a format based on identifiers.org.
+
+    Parameters
+    ----------
+    mat_dict: OrderedDict
+        An ordered dictionary having model attributes as keys and their
+        respective values represented as arrays, as the values. Annotations will
+        be inserted into this OrderdDict.
+    annotation_list: list[Dict]
+        A list of cobra annotations, in the form of a dictionary.
+    _d_replace: str
+        A string that points to the dictionary of converstions between MATLAB and
+        providers. Default D_MET_REV (for metabolite).
+    """
+    providers_used = set()
+    for i in range(len(annotation_list)):
+        if annotation_list[i]:
+            providers_used.update(annotation_list[i].keys())
+    providers_used = providers_used.intersection(D_REPLACE[_d_replace].keys())
+    providers_used = list(providers_used)
+    annotation_matlab = {prov: D_REPLACE[_d_replace][prov] for prov in providers_used}
+    empty_lists = [[""] * len(annotation_list) for x in annotation_matlab]
+    annotation_cells_to_be = dict(zip(annotation_matlab.values(), empty_lists))
+    for i in range(len(annotation_list)):
+        if annotation_list[i]:
+            for provider_key, v in annotation_list[i].items():
+                if provider_key == "pubmed":
+                    v = ", ".join(["PMID:" + annot for annot in v])
+                elif provider_key == "CHEBI":
+                    v = ", ".join(["CHEBI:" + annot for annot in v])
+                elif provider_key == "ec-code":
+                    v = " or ".join(v)
+                else:
+                    v = ", ".join(v)
+                if provider_key not in providers_used:
+                    continue
+                annotation_cells_to_be[annotation_matlab[provider_key]][i] = v
+    for annotation_key, _list in annotation_cells_to_be.items():
+        if annotation_key not in mat_dict:
+            mat_dict[annotation_key] = _cell(_list)
+
+
+def notes_to_mat(
```

Comment:
Okay.
This is again, not to lose information on round trips.
Something needs to happen with confidence scores, in order not to lose it.
The weird references go into notes from rxnReferences, and then they go back so they will be written in "rxnReferences". The whole _tmp_list2 section is for that.

Reply:
I mean, it would already lose information that does not fit the identifiers that can be mapped to matlab. If you write them to the Matlab notes it would not get lost but put in a different place (and one that is consistent with cobrapy/SBML). But the point of the notes field is that it should not contain information that you want to keep or have parsed.